queue-server.js

```diff
@@ -0,0 +1,47 @@
+const io = require('socket.io')(3001);
+
+let queue = {
+  flowerShop: [],
+  candyShop: [],
+};
+
+io.on('connection', (socket) => {
+  console.log('connected to', socket.id);
+
+  socket.on('subscribe', (payload) => {
+    if (payload === 'flower-shop') {
+      socket.join('flower-shop');
+    } else if (payload === 'candy-shop') {
+      socket.join('candy-shop');
+    }
+
+  });
+
+  socket.on('getAll', (payload) => {
+    if (payload === 'flower-shop') {
+      io.to('flower-shop').emit('orderQueue', queue.flowerShop);
+    } else if (payload === 'candy') {
+      io.to('candy-shop').emit('orderQueue', queue.candyShop);
+    }
+  });
+
+  socket.on('delivered', (payload) => {
+    if (payload.vendor === 'flower-shop') {
+      queue.flower.push(payload);
+      io.to('flower-shop').emit('delivered', payload);
+    } else if (payload.vendor === 'candy-shop') {
+      queue.candy.push(payload);

```

Comment:
Change 'flower'  and 'candy' to 'flowerShop' and 'candyShop' , because your trying to referred back to 'queue' which has the  the keys named 'flowerShop' and 'candyShop. Currently it going to 'queue' and not finding 'flower' and 'candy' because it does not existed.(also same thing with lines 40-43)